compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,285 @@
+            self.add_output_file(filename=filename)
+
+        # unpack the mesh parameters for the given resolution as set as
+        # attributes
+        self.nx, self.ny, self.dc = resolution_params[key].values()
+
+        self.resolution = resolution
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+
+        section = config['mismipplus']
+        nonperiodic = section.getboolean('nonperiodic')
+
+        # read the resolution from the .cfg file at runtime
+        resolution = section.getint('resolution')
+
+        # check if the resolution has been changed since the `compass setup`
+        # command was run
+        if self.resolution != resolution:
+            warnings.warn(f'Resolution was set at {self.resolution:2d}km when'
+                          f' `compass setup` was called. Since then the'
+                          f' resolution in the configuration file has been'
+                          f' changed to {resolution:2d}km. Changing resolution'
+                          f' at runtime is not supported. Change the'
+                          f' resolution value in the configuration file within'
+                          f' the python module and rerun the `compass setup`'
+                          f' command in order to create a mesh at a resolution'
+                          f' of {resolution:2d}km')
+
+        ds_mesh = make_planar_hex_mesh(nx=self.nx, ny=self.ny, dc=self.dc,
+                                       nonperiodic_x=nonperiodic,
+                                       nonperiodic_y=nonperiodic)
+
+        ds_mesh = cull(ds_mesh, logger=logger)
+        ds_mesh = convert(ds_mesh, logger=logger)
+
+        # custom function to shift the orgin for the MISMIP+ experiments
+        ds_mesh = center_trough(ds_mesh)
+
+        write_netcdf(ds_mesh, 'mpas_grid.nc')
+
+        # using `.get(...)`, instead of `.getint(...)` since variable needs
+        # to be string anyway for subprocess call
+        levels = section.get('levels')
+
+        args = ['create_landice_grid_from_generic_MPAS_grid.py',
+                '-i', 'mpas_grid.nc',
+                '-o', 'landice_grid.nc',
+                '-l', levels,
+                '--diri',
+                '--thermal']
+
+        check_call(args, logger)
+
+        make_graph_file(mesh_filename='landice_grid.nc',
+                        graph_filename='graph.info')
+
+        # create the initial conditions for the MISMIP+ spinup expr
+        _setup_MISMPPlus_IC(config, logger,
+                            filename='landice_grid.nc')
+
+
+def center_trough(ds_mesh):
+    """
+    Shift the orgin so that the bed trough is center about the Y-axis and the
+    X-axis is shifted all the way to the left.
+
+    Parameters
+    ----------
+    ds_mesh : xr.core.dataset.Datatset
+        The mesh to be shifted
+    """
+
+    # Why do you need unique grid points?
```

Comment:
@matthewhoffman  As you mentioned, for the translation `unique_xs.min()` is unnecessary since `xCell.min()` is equivalent, so I've just switched to the simpler option here. 

But, later on when we alter the `yEdge` length on the North/South boundaries  `unique_ys.min()` is still necessary because we need to alter both the boundary and the first interior edge on the North/South boundaries. `yCell.min()` can get us the boundary index but not the first interior, so I've left this approach there.